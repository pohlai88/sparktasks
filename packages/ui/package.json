{
  "name": "@sparktasks/ui",
  "version": "0.1.0",
  "private": true,
  "description": "SparkTask Railway UI Components - Fortune 500 Grade Design System",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.mjs",
      "require": "./dist/index.js"
    },
    "./styles": {
      "import": "./dist/styles.css"
    }
  },
  "files": [
    "dist/**"
  ],
  "scripts": {
    "build": "tsup",
    "dev": "tsup --watch",
    "lint": "eslint \"src/**/*.{ts,tsx}\"",
    "lint:fix": "eslint \"src/**/*.{ts,tsx}\" --fix",
    "type-check": "tsc --noEmit",
    "test": "vitest",
    "test:coverage": "vitest --coverage"
  },
  "dependencies": {
    "@radix-ui/react-slot": "^1.2.3",
    "@radix-ui/react-tabs": "^1.1.13",
    "@radix-ui/react-dialog": "^1.1.15",
    "@radix-ui/react-progress": "^1.1.7",
    "@radix-ui/react-tooltip": "^1.2.8",
    "@radix-ui/react-select": "^2.2.6",
    "@radix-ui/react-checkbox": "^1.3.3",
    "@radix-ui/react-label": "^2.1.7",
    "@radix-ui/react-separator": "^1.1.7",
    "@radix-ui/react-alert-dialog": "^1.1.15",
    "@radix-ui/react-dropdown-menu": "^2.1.16",
    "@radix-ui/react-icons": "^1.3.2",
    "class-variance-authority": "^0.7.1",
    "clsx": "^2.0.0",
    "tailwind-merge": "^2.6.0",
    "lucide-react": "^0.540.0",
    "framer-motion": "^11.18.2"
  },
  "peerDependencies": {
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
  "devDependencies": {
    "@types/react": "^18.3.0",
    "@types/react-dom": "^18.3.0",
    "typescript": "^5.0.0",
    "tsup": "^8.5.0",
    "eslint": "^8.57.1",
    "@typescript-eslint/eslint-plugin": "^8.40.0",
    "@typescript-eslint/parser": "^8.40.0",
    "vitest": "^3.2.4",
    "@vitest/coverage-v8": "^3.2.4",
    "jsdom": "^23.2.0",
    "@testing-library/react": "^13.4.0",
    "@testing-library/jest-dom": "^6.8.0",
    "tailwindcss": "^3.4.17"
  }
}
