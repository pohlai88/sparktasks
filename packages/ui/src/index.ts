export { Button, buttonVariants } from './button'
export { cn } from './utils'
