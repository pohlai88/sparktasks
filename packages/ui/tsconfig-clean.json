{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "composite": true,
    "outDir": "../../dist/packages/ui",
    "rootDir": "src",
    "jsx": "react-jsx",
    "declaration": true,
    "declarationMap": true,
    "noEmit": false,
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": [
    "src/**/*.ts",
    "src/**/*.tsx"
  ],
  "exclude": [
    "node_modules",
    "dist",
    "**/*.test.*",
    "**/*.spec.*"
  ]
}
