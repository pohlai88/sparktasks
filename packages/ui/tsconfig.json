{
  "compilerOptions": {
    "target": "ES2017",
    "lib": ["dom", "dom.iterable", "ES6"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "noEmit": false,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "bundler",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "react-jsx",
    "declaration": true,
    "declarationMap": true,
    "incremental": false,
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": [
    "src/**/*.ts",
    "src/**/*.tsx"
  ],
  "exclude": [
    "node_modules",
    "dist",
    "**/*.test.*",
    "**/*.spec.*"
  ]
}
