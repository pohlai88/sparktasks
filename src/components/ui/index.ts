// UI Component Library - SSOT Single Source of Truth
// All components use design tokens exclusively - zero hardcoded Tailwind classes

// Core Interactive Components
export { Button, IconButton } from './Button';
export { Card, CardHeader, CardTitle, CardContent, CardFooter } from './Card';
export { Badge, Chip } from './Badge';
export { Input, Textarea, Select } from './Input';

// Specialized Task Components  
export { PriorityBadge, StatusBadge, DueDateBadge } from './TaskBadges';
export { TagChip, TagList } from './TagChip';

// Layout & Navigation Components
export { Modal, ConfirmModal } from './Modal';
export { Tooltip } from './Tooltip';
export { Tabs, TabsList, TabsTrigger, TabsContent } from './Tabs';

// Form Components (Coming Soon)
// export { Checkbox } from './Checkbox';
// export { Radio } from './Radio'; 
// export { FormField } from './FormField';

// Feedback Components (Coming Soon)
// export { Alert } from './Alert';
// export { Skeleton } from './Skeleton';
// export { Spinner } from './Spinner';

// Data Display (Coming Soon)
// export { Table } from './Table';
// export { Avatar } from './Avatar';
// export { Progress } from './Progress';

// Re-export design tokens for components that need direct access
export { DESIGN_TOKENS, getPriorityStyles, getStatusStyles, getUrgencyStyles, getPriorityLabel, getStatusLabel } from '../../design/tokens';
