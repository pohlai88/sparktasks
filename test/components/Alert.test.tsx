import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { Alert, AlertAction, AlertGroup } from '../../src/components/ui/Alert';

describe('Alert Component', () => {
  // Basic rendering tests
  describe('Basic Rendering', () => {
    it('renders with default props', () => {
      render(<Alert>Test alert content</Alert>);
      expect(screen.getByRole('alert')).toBeInTheDocument();
      expect(screen.getByText('Test alert content')).toBeInTheDocument();
    });

    it('renders all variant types', () => {
      const variants = ['info', 'success', 'warning', 'error'] as const;

      variants.forEach(variant => {
        render(
          <Alert variant={variant} data-testid={`alert-${variant}`}>
            {variant} alert
          </Alert>
        );

        expect(screen.getByTestId(`alert-${variant}`)).toBeInTheDocument();
        expect(screen.getByText(`${variant} alert`)).toBeInTheDocument();
      });
    });

    it('renders all size variants', () => {
      const sizes = ['sm', 'md', 'lg'] as const;

      sizes.forEach(size => {
        render(
          <Alert size={size} data-testid={`alert-${size}`}>
            {size} alert
          </Alert>
        );

        expect(screen.getByTestId(`alert-${size}`)).toBeInTheDocument();
      });
    });

    it('renders with title', () => {
      render(<Alert title='Alert Title'>Alert content</Alert>);
      expect(screen.getByText('Alert Title')).toBeInTheDocument();
      expect(screen.getByText('Alert content')).toBeInTheDocument();
    });

    it('renders with custom icon', () => {
      const CustomIcon = <span data-testid='custom-icon'>ðŸ”¥</span>;
      render(<Alert icon={CustomIcon}>Alert with custom icon</Alert>);
      expect(screen.getByTestId('custom-icon')).toBeInTheDocument();
    });
  });

  // Dismissible functionality tests
  describe('Dismissible Functionality', () => {
    it('renders dismiss button when dismissible is true', () => {
      const onDismiss = vi.fn();
      render(
        <Alert dismissible onDismiss={onDismiss}>
          Dismissible alert
        </Alert>
      );

      expect(
        screen.getByRole('button', { name: /dismiss.*alert/i })
      ).toBeInTheDocument();
    });

    it('calls onDismiss when dismiss button is clicked', () => {
      const onDismiss = vi.fn();
      render(
        <Alert dismissible onDismiss={onDismiss}>
          Dismissible alert
        </Alert>
      );

      const dismissButton = screen.getByRole('button', {
        name: /dismiss.*alert/i,
      });
      fireEvent.click(dismissButton);

      expect(onDismiss).toHaveBeenCalledTimes(1);
    });

    it('handles keyboard interaction on dismiss button', () => {
      const onDismiss = vi.fn();
      render(
        <Alert dismissible onDismiss={onDismiss}>
          Dismissible alert
        </Alert>
      );

      const dismissButton = screen.getByRole('button', {
        name: /dismiss.*alert/i,
      });

      // Test Enter key
      fireEvent.keyDown(dismissButton, {
        key: 'Enter',
        preventDefault: vi.fn(),
        stopPropagation: vi.fn(),
      });
      expect(onDismiss).toHaveBeenCalledTimes(1);

      // Test Space key
      fireEvent.keyDown(dismissButton, {
        key: ' ',
        preventDefault: vi.fn(),
        stopPropagation: vi.fn(),
      });
      expect(onDismiss).toHaveBeenCalledTimes(2);
    });

    it('does not render dismiss button when dismissible is false', () => {
      render(<Alert dismissible={false}>Non-dismissible alert</Alert>);
      expect(
        screen.queryByRole('button', { name: /dismiss.*alert/i })
      ).not.toBeInTheDocument();
    });

    it('does not render dismiss button when onDismiss is not provided', () => {
      render(<Alert dismissible>Alert without onDismiss</Alert>);
      expect(
        screen.queryByRole('button', { name: /dismiss.*alert/i })
      ).not.toBeInTheDocument();
    });
  });

  // Actions rendering tests
  describe('Actions Rendering', () => {
    it('renders actions when provided', () => {
      const actions = (
        <div>
          <AlertAction variant='primary'>Primary Action</AlertAction>
          <AlertAction variant='secondary'>Secondary Action</AlertAction>
        </div>
      );

      render(<Alert actions={actions}>Alert with actions</Alert>);

      expect(screen.getByText('Primary Action')).toBeInTheDocument();
      expect(screen.getByText('Secondary Action')).toBeInTheDocument();
    });

    it('does not render actions section when actions are not provided', () => {
      render(<Alert>Alert without actions</Alert>);

      // The actions container should not exist
      expect(screen.queryByText('Primary Action')).not.toBeInTheDocument();
    });

    it('AlertAction component renders correctly', () => {
      render(
        <AlertAction data-testid='alert-action' onClick={vi.fn()}>
          Test Action
        </AlertAction>
      );

      expect(screen.getByTestId('alert-action')).toBeInTheDocument();
      expect(screen.getByText('Test Action')).toBeInTheDocument();
    });
  });

  // ARIA labels and accessibility tests
  describe('ARIA Labels and Accessibility', () => {
    it('generates semantic ARIA label without title', () => {
      render(<Alert variant='warning'>Warning content</Alert>);

      const alert = screen.getByRole('alert');
      expect(alert).toHaveAttribute('aria-label', 'warning alert');
    });

    it('generates semantic ARIA label with title', () => {
      render(
        <Alert variant='error' title='Critical Error'>
          Error content
        </Alert>
      );

      const alert = screen.getByRole('alert');
      expect(alert).toHaveAttribute(
        'aria-label',
        'error alert: Critical Error'
      );
    });

    it('uses custom aria-label when provided', () => {
      render(<Alert aria-label='Custom alert label'>Alert content</Alert>);

      const alert = screen.getByRole('alert');
      expect(alert).toHaveAttribute('aria-label', 'Custom alert label');
    });

    it('has correct ARIA attributes', () => {
      render(<Alert>Alert content</Alert>);

      const alert = screen.getByRole('alert');
      expect(alert).toHaveAttribute('aria-live', 'polite');
      expect(alert).toHaveAttribute('aria-atomic', 'true');
    });

    it('includes screen reader content for complex alerts', () => {
      const multiLineContent = 'Multi-line content\nwith line breaks';

      render(
        <Alert
          variant='success'
          title='Success Title'
          dismissible
          onDismiss={vi.fn()}
          actions={<AlertAction>Action</AlertAction>}
        >
          {multiLineContent}
        </Alert>
      );

      // Screen reader content should be present but visually hidden
      expect(screen.getByText(/Alert type: success/)).toBeInTheDocument();
      expect(screen.getByText(/Title: Success Title/)).toBeInTheDocument();
      expect(screen.getByText(/2 lines of content/)).toBeInTheDocument();
      expect(
        screen.getByText(/Press dismiss button to close alert/)
      ).toBeInTheDocument();
      expect(
        screen.getByText(/Additional actions available/)
      ).toBeInTheDocument();
    });

    it('has accessible dismiss button label', () => {
      const onDismiss = vi.fn();
      render(
        <Alert variant='info' dismissible onDismiss={onDismiss}>
          Alert
        </Alert>
      );

      const dismissButton = screen.getByRole('button', {
        name: 'Dismiss info alert',
      });
      expect(dismissButton).toBeInTheDocument();
    });
  });

  // AlertGroup component tests
  describe('AlertGroup Component', () => {
    it('renders multiple alerts with correct spacing', () => {
      render(
        <AlertGroup spacing='normal' data-testid='alert-group'>
          <Alert>First alert</Alert>
          <Alert>Second alert</Alert>
          <Alert>Third alert</Alert>
        </AlertGroup>
      );

      const group = screen.getByTestId('alert-group');
      expect(group).toBeInTheDocument();
      expect(group).toHaveAttribute('role', 'group');
      expect(group).toHaveAttribute('aria-label', 'Alert notifications');

      expect(screen.getByText('First alert')).toBeInTheDocument();
      expect(screen.getByText('Second alert')).toBeInTheDocument();
      expect(screen.getByText('Third alert')).toBeInTheDocument();
    });

    it('applies different spacing variants', () => {
      const spacings = ['tight', 'normal', 'relaxed'] as const;

      spacings.forEach(spacing => {
        render(
          <AlertGroup spacing={spacing} data-testid={`group-${spacing}`}>
            <Alert>Alert in {spacing} group</Alert>
          </AlertGroup>
        );

        expect(screen.getByTestId(`group-${spacing}`)).toBeInTheDocument();
      });
    });
  });

  // RTL and internationalization tests
  describe('RTL and Internationalization', () => {
    it('renders correctly in RTL mode', () => {
      render(
        <div dir='rtl'>
          <Alert title='Ø¹Ù†ÙˆØ§Ù† Ø§Ù„ØªÙ†Ø¨ÙŠÙ‡' dismissible onDismiss={vi.fn()}>
            Ù…Ø­ØªÙˆÙ‰ Ø§Ù„ØªÙ†Ø¨ÙŠÙ‡
          </Alert>
        </div>
      );

      expect(screen.getByText('Ø¹Ù†ÙˆØ§Ù† Ø§Ù„ØªÙ†Ø¨ÙŠÙ‡')).toBeInTheDocument();
      expect(screen.getByText('Ù…Ø­ØªÙˆÙ‰ Ø§Ù„ØªÙ†Ø¨ÙŠÙ‡')).toBeInTheDocument();
      expect(
        screen.getByRole('button', { name: /dismiss.*alert/i })
      ).toBeInTheDocument();
    });

    it('handles long titles gracefully', () => {
      const longTitle =
        'This is a very long alert title that should handle text wrapping and overflow gracefully in different viewport sizes';

      render(<Alert title={longTitle}>Alert content</Alert>);
      expect(screen.getByText(longTitle)).toBeInTheDocument();
    });

    it('handles long content gracefully', () => {
      const longContent =
        'This is a very long alert content that spans multiple lines and should handle text wrapping, overflow, and different viewport sizes gracefully while maintaining readability and accessibility.';

      render(<Alert>{longContent}</Alert>);
      expect(screen.getByText(longContent)).toBeInTheDocument();
    });
  });

  // Visual regression and snapshot tests
  describe('Visual Regression', () => {
    it('maintains consistent visual structure', () => {
      const { container } = render(
        <Alert
          variant='warning'
          size='md'
          title='Warning Title'
          dismissible
          onDismiss={vi.fn()}
          actions={
            <div>
              <AlertAction variant='primary'>Primary</AlertAction>
              <AlertAction variant='secondary'>Secondary</AlertAction>
            </div>
          }
        >
          Warning message content with multiple lines and detailed information
        </Alert>
      );

      // Test that the DOM structure matches expectations
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders minimal alert correctly', () => {
      const { container } = render(<Alert>Simple alert</Alert>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders complex alert group correctly', () => {
      const { container } = render(
        <AlertGroup spacing='normal'>
          <Alert variant='info' title='Info'>
            Information message
          </Alert>
          <Alert variant='success' dismissible onDismiss={vi.fn()}>
            Success message
          </Alert>
          <Alert
            variant='error'
            title='Error'
            actions={<AlertAction>Retry</AlertAction>}
          >
            Error message with action
          </Alert>
        </AlertGroup>
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  // Error boundary and edge cases
  describe('Edge Cases and Error Handling', () => {
    it('handles empty content gracefully', () => {
      render(<Alert>{''}</Alert>);
      expect(screen.getByRole('alert')).toBeInTheDocument();
    });

    it('handles null children', () => {
      render(<Alert>{null}</Alert>);
      expect(screen.getByRole('alert')).toBeInTheDocument();
    });

    it('handles undefined title', () => {
      render(<Alert>Content</Alert>);
      expect(screen.getByText('Content')).toBeInTheDocument();
    });

    it('forwards ref correctly', () => {
      const ref = React.createRef<HTMLDivElement>();
      render(<Alert ref={ref}>Alert with ref</Alert>);

      expect(ref.current).toBeInstanceOf(HTMLDivElement);
      expect(ref.current).toHaveAttribute('role', 'alert');
    });

    it('handles custom className correctly', () => {
      render(
        <Alert className='custom-alert-class'>Alert with custom class</Alert>
      );

      const alert = screen.getByRole('alert');
      expect(alert).toHaveClass('custom-alert-class');
    });

    it('spreads additional props correctly', () => {
      render(
        <Alert data-custom='custom-value' id='custom-alert'>
          Alert with custom props
        </Alert>
      );

      const alert = screen.getByRole('alert');
      expect(alert).toHaveAttribute('data-custom', 'custom-value');
      expect(alert).toHaveAttribute('id', 'custom-alert');
    });
  });
});
