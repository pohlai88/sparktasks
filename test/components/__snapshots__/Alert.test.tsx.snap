// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert Component > Visual Regression > maintains consistent visual structure 1`] = `
<div
  aria-atomic="true"
  aria-label="warning alert: Warning Title"
  aria-live="polite"
  class="relative rounded-lg transition-all duration-200 ease-out motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary-600 focus-visible:ring-offset-2 forced-colors:outline forced-colors:outline-2 border border-warning-200 dark:border-warning-800 bg-warning-50 dark:bg-warning-950/30 text-warning-900 dark:text-warning-100 p-4"
  role="alert"
>
  <div
    class="flex gap-3"
  >
    <div
      class="shrink-0 text-warning-600 dark:text-warning-400 w-5 h-5"
    >
      <svg
        aria-hidden="true"
        class="w-5 h-5"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 20 20"
      >
        <path
          clip-rule="evenodd"
          d="M8.485 2.495c.673-1.167 2.357-1.167 3.03 0l6.28 10.875c.673 1.167-.17 2.625-1.516 2.625H3.72c-1.347 0-2.189-1.458-1.515-2.625L8.485 2.495zM10 5a.75.75 0 01.75.75v3.5a.75.75 0 01-1.5 0v-3.5A.75.75 0 0110 5zm0 9a1 1 0 100-2 1 1 0 000 2z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="flex-1 min-w-0"
    >
      <h3
        class="font-medium leading-tight mb-1 text-warning-900 dark:text-warning-100 text-base font-medium leading-relaxed text-slate-800 dark:text-slate-200"
      >
        Warning Title
      </h3>
      <div
        class="leading-relaxed text-warning-800 dark:text-warning-200 text-sm leading-relaxed text-slate-700 dark:text-slate-300"
      >
        Warning message content with multiple lines and detailed information
      </div>
      <div
        class="mt-3"
      >
        <div>
          <button
            class="inline-flex items-center justify-center px-3 py-1.5 text-sm font-medium rounded-md transition-colors duration-200 motion-reduce:transition-none focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500 dark:focus:ring-primary-400 dark:focus:ring-offset-slate-900 disabled:opacity-50 disabled:cursor-not-allowed bg-primary-600 text-white hover:bg-primary-700 active:bg-primary-800"
          >
            Primary
          </button>
          <button
            class="inline-flex items-center justify-center px-3 py-1.5 text-sm font-medium rounded-md transition-colors duration-200 motion-reduce:transition-none focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500 dark:focus:ring-primary-400 dark:focus:ring-offset-slate-900 disabled:opacity-50 disabled:cursor-not-allowed bg-transparent text-current border border-current hover:bg-current/10 active:bg-current/20"
          >
            Secondary
          </button>
        </div>
      </div>
    </div>
    <div
      class="shrink-0"
    >
      <button
        aria-label="Dismiss warning alert"
        class="p-1 rounded-md hover:bg-slate-100 dark:hover:bg-slate-700 transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary-600 focus-visible:ring-offset-2 forced-colors:outline forced-colors:outline-2 text-warning-500 hover:text-warning-700 dark:text-warning-400 dark:hover:text-warning-200"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="w-4 h-4"
          fill="none"
          focusable="false"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    class="sr-only"
  >
    Alert type: 
    warning
    .
     Title: Warning Title.
     Press dismiss button to close alert.
     Additional actions available.
  </div>
</div>
`;

exports[`Alert Component > Visual Regression > renders complex alert group correctly 1`] = `
<div
  aria-label="Alert notifications"
  class="space-y-3"
  role="group"
>
  <div
    aria-atomic="true"
    aria-label="info alert: Info"
    aria-live="polite"
    class="relative rounded-lg transition-all duration-200 ease-out motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary-600 focus-visible:ring-offset-2 forced-colors:outline forced-colors:outline-2 border border-primary-200 dark:border-primary-800 bg-primary-50 dark:bg-primary-950/30 text-primary-900 dark:text-primary-100 p-4"
    role="alert"
  >
    <div
      class="flex gap-3"
    >
      <div
        class="shrink-0 text-primary-600 dark:text-primary-400 w-5 h-5"
      >
        <svg
          aria-hidden="true"
          class="w-5 h-5"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 20 20"
        >
          <path
            clip-rule="evenodd"
            d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a.75.75 0 000 1.5h.253a.25.25 0 01.244.304l-.459 2.066A1.75 1.75 0 0010.747 15H11a.75.75 0 000-1.5h-.253a.25.25 0 01-.244-.304l.459-2.066A1.75 1.75 0 009.253 9H9z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="flex-1 min-w-0"
      >
        <h3
          class="font-medium leading-tight mb-1 text-primary-900 dark:text-primary-100 text-base font-medium leading-relaxed text-slate-800 dark:text-slate-200"
        >
          Info
        </h3>
        <div
          class="leading-relaxed text-primary-800 dark:text-primary-200 text-sm leading-relaxed text-slate-700 dark:text-slate-300"
        >
          Information message
        </div>
      </div>
    </div>
    <div
      class="sr-only"
    >
      Alert type: 
      info
      .
       Title: Info.
    </div>
  </div>
  <div
    aria-atomic="true"
    aria-label="success alert"
    aria-live="polite"
    class="relative rounded-lg transition-all duration-200 ease-out motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary-600 focus-visible:ring-offset-2 forced-colors:outline forced-colors:outline-2 border border-success-200 dark:border-success-800 bg-success-50 dark:bg-success-950/30 text-success-900 dark:text-success-100 p-4"
    role="alert"
  >
    <div
      class="flex gap-3"
    >
      <div
        class="shrink-0 text-success-600 dark:text-success-400 w-5 h-5"
      >
        <svg
          aria-hidden="true"
          class="w-5 h-5"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 20 20"
        >
          <path
            clip-rule="evenodd"
            d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.857-9.809a.75.75 0 00-1.214-.882l-3.236 4.53L8.25 10.5a.75.75 0 00-1.06 1.061l1.5 1.5a.75.75 0 001.137-.089l4-5.5z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="flex-1 min-w-0"
      >
        <div
          class="leading-relaxed text-success-800 dark:text-success-200 text-sm leading-relaxed text-slate-700 dark:text-slate-300"
        >
          Success message
        </div>
      </div>
      <div
        class="shrink-0"
      >
        <button
          aria-label="Dismiss success alert"
          class="p-1 rounded-md hover:bg-slate-100 dark:hover:bg-slate-700 transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary-600 focus-visible:ring-offset-2 forced-colors:outline forced-colors:outline-2 text-success-500 hover:text-success-700 dark:text-success-400 dark:hover:text-success-200"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="w-4 h-4"
            fill="none"
            focusable="false"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="sr-only"
    >
      Alert type: 
      success
      .
       Press dismiss button to close alert.
    </div>
  </div>
  <div
    aria-atomic="true"
    aria-label="error alert: Error"
    aria-live="polite"
    class="relative rounded-lg transition-all duration-200 ease-out motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary-600 focus-visible:ring-offset-2 forced-colors:outline forced-colors:outline-2 border border-error-200 dark:border-error-800 bg-error-50 dark:bg-error-950/30 text-error-900 dark:text-error-100 p-4"
    role="alert"
  >
    <div
      class="flex gap-3"
    >
      <div
        class="shrink-0 text-error-600 dark:text-error-400 w-5 h-5"
      >
        <svg
          aria-hidden="true"
          class="w-5 h-5"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 20 20"
        >
          <path
            clip-rule="evenodd"
            d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zM8.28 7.22a.75.75 0 00-1.06 1.06L8.94 10l-1.72 1.72a.75.75 0 101.06 1.06L10 11.06l1.72 1.72a.75.75 0 101.06-1.06L11.06 10l1.72-1.72a.75.75 0 00-1.06-1.06L10 8.94 8.28 7.22z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="flex-1 min-w-0"
      >
        <h3
          class="font-medium leading-tight mb-1 text-error-900 dark:text-error-100 text-base font-medium leading-relaxed text-slate-800 dark:text-slate-200"
        >
          Error
        </h3>
        <div
          class="leading-relaxed text-error-800 dark:text-error-200 text-sm leading-relaxed text-slate-700 dark:text-slate-300"
        >
          Error message with action
        </div>
        <div
          class="mt-3"
        >
          <button
            class="inline-flex items-center justify-center px-3 py-1.5 text-sm font-medium rounded-md transition-colors duration-200 motion-reduce:transition-none focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500 dark:focus:ring-primary-400 dark:focus:ring-offset-slate-900 disabled:opacity-50 disabled:cursor-not-allowed bg-transparent text-current border border-current hover:bg-current/10 active:bg-current/20"
          >
            Retry
          </button>
        </div>
      </div>
    </div>
    <div
      class="sr-only"
    >
      Alert type: 
      error
      .
       Title: Error.
       Additional actions available.
    </div>
  </div>
</div>
`;

exports[`Alert Component > Visual Regression > renders minimal alert correctly 1`] = `
<div
  aria-atomic="true"
  aria-label="info alert"
  aria-live="polite"
  class="relative rounded-lg transition-all duration-200 ease-out motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary-600 focus-visible:ring-offset-2 forced-colors:outline forced-colors:outline-2 border border-primary-200 dark:border-primary-800 bg-primary-50 dark:bg-primary-950/30 text-primary-900 dark:text-primary-100 p-4"
  role="alert"
>
  <div
    class="flex gap-3"
  >
    <div
      class="shrink-0 text-primary-600 dark:text-primary-400 w-5 h-5"
    >
      <svg
        aria-hidden="true"
        class="w-5 h-5"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 20 20"
      >
        <path
          clip-rule="evenodd"
          d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a.75.75 0 000 1.5h.253a.25.25 0 01.244.304l-.459 2.066A1.75 1.75 0 0010.747 15H11a.75.75 0 000-1.5h-.253a.25.25 0 01-.244-.304l.459-2.066A1.75 1.75 0 009.253 9H9z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="flex-1 min-w-0"
    >
      <div
        class="leading-relaxed text-primary-800 dark:text-primary-200 text-sm leading-relaxed text-slate-700 dark:text-slate-300"
      >
        Simple alert
      </div>
    </div>
  </div>
  <div
    class="sr-only"
  >
    Alert type: 
    info
    .
  </div>
</div>
`;
